components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -364,6 +390,65 @@ public OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumer
         return OAuthUtil.buildConsumerAppDTO(app);
     }
 
+
+    private Optional<AuthenticatedUser> getLoggedInUser(String tenantDomain) {
+
+        String tenantAwareLoggedInUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        return Optional.ofNullable(tenantAwareLoggedInUsername)
+                .filter(StringUtils::isNotEmpty)
+                .map(username -> buildAuthenticatedUser(username, tenantDomain));
+
+    }
+
+    private static Map<String, Object> buildSPData(OAuthAppDO app) {
+
+        if (app == null) {
+            return new HashMap<>();
+        }
+        Gson gson = new Gson();
+        String oauthApp = maskSPData(app);
+        return gson.fromJson(oauthApp, new TypeToken<Map<String, Object>>() {
+        }.getType());
+    }
+
+    private static String maskSPData(OAuthAppDO oAuthAppDO) {
+
+        if (oAuthAppDO == null) {
+            return StringUtils.EMPTY;
+        }
+        try {
+            JSONObject oauthAppJSONObject =
+                    new JSONObject(new ObjectMapper().writeValueAsString(oAuthAppDO));
+            maskClientSecret(oauthAppJSONObject);
+            maskAppOwnerUsername(oauthAppJSONObject);
+            return oauthAppJSONObject.toString();
+        } catch (JsonProcessingException | IdentityException e) {
+            LOG.error("Error while converting service provider object to json.");
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private static void maskAppOwnerUsername(JSONObject oauthAppJSONObject) throws IdentityException {
+
+        JSONObject appOwner = oauthAppJSONObject.optJSONObject("appOwner");
+        if (!LoggerUtils.isLogMaskingEnable || appOwner == null) {
+            return;
+        }
+        String username = (String) appOwner.get("userName");
+        if (StringUtils.isNotBlank(username)) {
+            appOwner.put("userName", LoggerUtils.getMaskedContent(username));
+        }
+    }
+
+    private static void maskClientSecret(JSONObject oauthApp) {
+
+        if (oauthApp.get("oauthConsumerSecret") == null) {
+            return;
+        }
+        String secret = oauthApp.get("oauthConsumerSecret").toString();
+        oauthApp.put("oauthConsumerSecret", LoggerUtils.getMaskedContent(secret));
+    }
```

Comment:
Can't we merge these two and have a single method like `maskField`?

eg: maskField(JSONObject jsonObject, String... path)

Reply:
and call that in the maskSPData method like;
maskField(oauthAppJSONObject, "appOwner", "userName");
    maskField(oauthAppJSONObject, "oauthConsumerSecret");

Reply:
We can have a single method, but still the logic will be different. For example, clientSecret is available as first class attribute in the path. But username is available as "appOwner:{"username":"xxx}". The path of the masking object is not a flat one. In some cases, it is nested.